src/game/client/maputil.cpp

```diff
@@ -67,11 +94,848 @@ Utf8String MapCache::Get_User_Map_Dir()
+{
+    s_worldDict = input.Read_Dict();
+    return true;
+}
+
+bool Parse_Object_Data_Chunk(DataChunkInput &input, DataChunkInfo *info, void *data)
+{
+    bool b = info->version >= 2;
+    Coord3D loc;
+    loc.x = input.Read_Real32();
+    loc.y = input.Read_Real32();
+    loc.z = input.Read_Real32();
+
+    if (info->version <= 2) {
+        loc.z = 0.0f;
+    }
+
+    float angle = input.Read_Real32();
+    int flags = input.Read_Int32();
+    Utf8String name = input.Read_AsciiString();
+    Dict props;
+
+    if (b) {
+        props = input.Read_Dict();
+    }
+
+    ThingTemplate *tmplate = nullptr;
+
+    MapObject *object = new MapObject(loc, name, angle, flags, &props, g_theThingFactory->Find_Template(name, false));
+
+    if (object->Get_Properties()->Get_Type(g_waypointIDKey) == Dict::DICT_INT) {
+        object->Set_Is_Waypoint();
+        (*s_waypoints)[object->Get_Waypoint_Name()] = loc;
+    } else if (object->Get_Thing_Template() != nullptr && object->Get_Thing_Template()->Is_KindOf(KINDOF_TECH_BUILDING)) {
+        s_techPositions.push_back(loc);
+    } else if (object->Get_Thing_Template() != nullptr) {
+        if (object->Get_Thing_Template()->Is_KindOf(KINDOF_SUPPLY_SOURCE_ON_PREVIEW)) {
+            s_supplyPositions.push_back(loc);
+        }
+    }
+
+    object->Delete_Instance();
+    return true;
+}
+
+bool Parse_Objects_Data_Chunk(DataChunkInput &input, DataChunkInfo *info, void *data)
+{
+    input.m_currentObject = nullptr;
+    input.Register_Parser("Object", info->label, Parse_Object_Data_Chunk, nullptr);
+    return input.Parse(data);
+}
+
+bool Load_Map(Utf8String name)
+{
+    Utf8String str;
+    int len = 0;
+    char buf[PATH_MAX];
+    strcpy(buf, name.Str());
+    len = strlen(buf);
+    char buf2[PATH_MAX];
+
+    if (len >= 4) {
+        memset(buf2, 0, PATH_MAX);
+        strncpy(buf2, buf, len - 4);
+    }
+
+    CachedFileInputStream stream;
+    str = name;
+
+    if (stream.Open(str)) {
+        DataChunkInput input(&stream);
+        s_waypoints = new WaypointMap();
+        input.Register_Parser("HeightMapData", Utf8String::s_emptyString, Parse_Size_Only_In_Chunk, nullptr);
+        input.Register_Parser("WorldInfo", Utf8String::s_emptyString, Parse_World_Dict_Data_Chunk, nullptr);
+        input.Register_Parser("ObjectsList", Utf8String::s_emptyString, Parse_Objects_Data_Chunk, nullptr);
+
+        if (!input.Parse(nullptr)) {
+            throw CODE_05;
+        }
+
+        s_mapDX = s_width - 2 * s_borderSize;
+        s_mapDY = s_height - 2 * s_borderSize;
+        return true;
+    } else {
+        return false;
+    }
+}
+
+unsigned int Calc_CRC(Utf8String dir, Utf8String name)
+{
+    CRC crc;
+    crc.Clear();
+
+    Utf8String str;
+    char buf[PATH_MAX];
+    strcpy(buf, name.Str());
+    int len = strlen(buf);
+    char buf2[PATH_MAX];
+
+    if (len >= 4) {
+        memset(buf2, 0, PATH_MAX);
+        strncpy(buf2, buf, len - 4);
+    }
```

Comment:
Same as function above. What are buf and buf2 for? They do nothing.

Reply:
No idea, original does it this way.